velox/type/DecimalUtil.h

```diff
@@ -32,5 +32,34 @@ class DecimalUtil {
   /// Helper function to convert a decimal value to string.
   template <typename T>
   static std::string toString(const T& value, const TypePtr& type);
+
+  inline static int128_t rescaleWithRoundUp(
+      const int128_t& unscaledValue,
+      const uint8_t fromScale,
+      const std::pair<int, int>& toPrecisionScale) {
+    auto rescaledValue = unscaledValue;
+    auto scaleDifference = toPrecisionScale.second - fromScale;
+    if (scaleDifference >= 0) {
+      rescaledValue *= DecimalUtil::kPowersOfTen[scaleDifference];
+    } else {
+      scaleDifference = -scaleDifference;
+      rescaledValue =
+          unscaledValue / DecimalUtil::kPowersOfTen[scaleDifference];
+      int128_t remainder =
+          unscaledValue % DecimalUtil::kPowersOfTen[scaleDifference];
+      if (unscaledValue >= 0 &&
+          remainder > DecimalUtil::kPowersOfTen[scaleDifference] / 2) {
+        rescaledValue++;
+      } else if (remainder < -DecimalUtil::kPowersOfTen[scaleDifference] / 2) {
+        rescaledValue--;
+      }
+    }
+    // Check overflowing
```

Comment:
I think it will be nice from an API standpoint to pass `nullOnError` here and return an optional value or throw.